webapp/mixins/scrollToAnchor.js

```diff
@@ -0,0 +1,32 @@
+export function scrollToAnchor(anchor) {
+  if (!anchor) return
+  if (!window || !document) {
+    return
+  }
+  const container = document.querySelector(anchor)
+  if (container) {
+    const { top } = container.getBoundingClientRect()
+    setTimeout(() => {
+      // we have to set a small timeout to ensure this part comes after nuxt
+      // scrollBehaviour: https://nuxtjs.org/api/configuration-router/#scrollbehavior
+      window.scroll({
+        top,
+        left: 0,
+        behavior: 'smooth',
+      })
+    }, 250)
+  }
+}
+
+export default {
```

Comment:
@roschaefer there is a small flaw in this mixin. With the current implementation, `scrollToAnchor` will be called by each component on the page no matter what is the value of the `hash`. In our case with 10 comments on the page and `hash = '#comments'` we'll try to scroll 11 times. 
Please see suggestion below on how it can be solved. Not sure though if any of the app patterns are violated with that approach,

```suggestion
export default function(checkAnchor = () => true) {
  return {
    watch: {
      $route(to, from) {
        const anchor = to && to.hash
        checkAnchor.call(this, anchor) && scrollToAnchor(anchor)
      },
    },
    mounted() {
      const anchor = this.$route && this.$route.hash
      checkAnchor.call(this, anchor) && scrollToAnchor(anchor)
    },
  }
}
```

And to consume the mixin we'd use something like:
```javascript
// in the component
...
mixins: [scrollToAnchor(anchor => anchor === '#comments')
```